### ISSUE TITLE
Search iterators are broken

### ISSUE THREAD

**Issue by bakurin** [elastic/elasticsearch-php#1145]
### Search iterators are broken

It seems that after #1044 `SearchResponseIterator` is broken: a deprecation error is thrown on an attempt to iterate the search result.

### Code snippet of problem

```php
$res = new SearchHitIterator(new SearchResponseIterator($client, $params));
foreach ($res as $hit) {
    //
}
```

### System details

- Operating System: Alpine Linux
- PHP Version: 8.0
- ES-PHP client version: 7.12
- Elasticsearch version: 7.4

There is my PR #1134 with the suggested fix

**Comment by ezimuel** [comment]
@bakurin thanks for opening this issue and for the PR #1134. I just reviewed it.

**Comment by ezimuel** [comment]
Merged in #1134
